packages/radix-vue/src/DropdownMenu/DropdownMenuSubContent.vue

```diff
@@ -41,13 +36,28 @@ const props = withDefaults(defineProps<DropdownMenuSubContentProps>(), {
   avoidCollisions: true,
 });
 
+defineExpose({
+  providesCollection: true,
+});
+
+const rootInjectedValue = injectSafely<DropdownMenuProvideValue>(
+  DROPDOWN_MENU_INJECTION_KEY,
+  Components.ROOT
+);
+
```

Comment:
I like this `inhjectSafely` utility. However I think we can improve it.


Im thinking since we already getting the `DROPDOWN_MENU_INJECTION_KEY` from RootComponent, can we someone use that key as an identifier instead of exposing all the `componentName`, and importing it from`constants`?

This way, we don't need to modify much for other components that are having this `Injection` warning, making migration to this `injectedSafely` so much easier.

Reply:
Brilliant! I have committed the changes.

There's a small change that must be noted. Instead of the root component name I have defined the expose `isComponentRoot` to break earlier the loop and can be shared with `injectCollection` (which before was using `isRoot`).

```ts
defineExpose({
  isComponentRoot: true,
  providerKey: SOME_ROOT_PROVIDER_KEY,
})
```

Reply:
hmm.. that's not quite what I was looking for.

I've tested with the following, by checking if the `injectionKey` was provided in the `getCurrentInstance`, and it works quite well.


```ts
export function injectSafely<T>(injectionKey: InjectionKey<T>) {
  if (isSSR) return {} as T; // Suppress Nuxt SSR warning

  const instance = getCurrentInstance();
  // @ts-ignore provides type was remove from type `ComponentInternalInstance` due to internal feature
  // It's fine for us to use as a injection checker
  if (instance?.provides?.[injectionKey]) return inject(injectionKey);
  return {} as T;
}
```

Reply:
I know what you mean, but as per Vue docs, relying on internal properties in production is not safe nor advised because those properties can change between Vue versions without further notice.

There's a reason why the Vue team didn't include them in the types and don't want us to rely on them, and that's exactly why I decided to go with the `defineExpose` approach instead of the above one: it's a bit more verbose I know that, but at least is safe.

I believe that when a package is meant to be used by thousands of people, we just cannot suppress an explicit core warning with `@ts-ignore` or to say: "warnings are fine since they're not displayed in production". People's is simply going to have problems during the first minutes or development and choose something else.

I did my best to help you out, but it doesn't seem we're on the same page on this matter. And that's perfectly fine. 

I personally won't keep working on this PR unless thing are done safely. If you're not ok with that, feel free to close it or to adapt the code as you see fit.

Reply:
Thanks for the thorough explanation and reasoning with this approach. Your finding was super valuable! 

Sorry if my suggestion or comment made you fed up, I'm trying to understand, and find the best solution for this issue here.

Reply:
Soooo.. umm.. anything on this? @zernonia